Dalamud/Interface/Internal/UiDebug2/Browsing/AddonTree.cs

```diff
@@ -0,0 +1,242 @@
+using static Dalamud.Utility.Util;
+
+namespace Dalamud.Interface.Internal.UiDebug2.Browsing;
+
+/// <summary>
+/// A class representing an <see cref="AtkUnitBase"/>, allowing it to be browsed within an ImGui window.
+/// </summary>
+public unsafe partial class AddonTree : IDisposable
+{
+    private readonly nint initialPtr;
+
+    private AddonPopoutWindow? window;
+
+    private AddonTree(string name, nint ptr)
+    {
+        this.AddonName = name;
+        this.initialPtr = ptr;
+        this.PopulateFieldNames(ptr);
+    }
+
+    /// <summary>
+    /// Gets the name of the addon this tree represents.
+    /// </summary>
+    internal string AddonName { get; init; }
+
+    /// <summary>
+    /// Gets or sets a collection of trees representing nodes within this addon.
+    /// </summary>
+    internal Dictionary<nint, ResNodeTree> NodeTrees { get; set; } = [];
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        foreach (var nodeTree in this.NodeTrees)
+        {
+            nodeTree.Value.Dispose();
+        }
+
+        this.NodeTrees.Clear();
+        this.FieldNames.Clear();
+        AddonTrees.Remove(this.AddonName);
+        if (this.window != null && PopoutWindows.Windows.Contains(this.window))
+        {
+            PopoutWindows.RemoveWindow(this.window);
+            this.window?.Dispose();
+        }
+
+        GC.SuppressFinalize(this);
+    }
+
+    /// <summary>
+    /// Gets an instance of <see cref="AddonTree"/> for the given addon name (or creates one if none are found).
+    /// The tree can then be drawn within the Addon Inspector and browsed.
+    /// </summary>
+    /// <param name="name">The name of the addon.</param>
+    /// <returns>The <see cref="AddonTree"/> for the named addon. Returns null if it does not exist, or if it is not at the expected address.</returns>
+    internal static AddonTree? GetOrCreate(string? name)
+    {
+        if (name == null)
+        {
+            return null;
+        }
+
+        try
+        {
+            var ptr = GameGui.GetAddonByName(name);
+
+            if ((AtkUnitBase*)ptr != null)
+            {
+                if (AddonTrees.TryGetValue(name, out var tree))
+                {
+                    if (tree.initialPtr == ptr)
+                    {
+                        return tree;
+                    }
+
+                    tree.Dispose();
+                }
+
+                var newTree = new AddonTree(name, ptr);
+                AddonTrees.Add(name, newTree);
+                return newTree;
+            }
+        }
+        catch
+        {
+            Log.Warning("Couldn't get addon!");
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Draws this AddonTree within a window.
+    /// </summary>
+    internal void Draw()
+    {
+        if (!this.ValidateAddon(out var addon))
+        {
+            return;
+        }
+
+        var isVisible = addon->IsVisible;
+
+        ImGui.Text($"{this.AddonName}");
```

Comment:
`TextUnformatted()` should always be used over `Text()`

The provided string formatting that `Text()` offers isn't useable in C#, but will instantly lead to crashes the moment it encounters a % symbol